infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+        assertThat(compareBaseHeaderFields(request.getHeader(), response.getHeader())).isTrue();
+        assertThat(response.getPayload().hasBootstrapSequencerResponse()).isTrue();
+        assertThat(response.getPayload().getBootstrapSequencerResponse().getRespType())
+                .isEqualTo(Sequencer.BootstrapSequencerResponseMsg.Type.ACK);
+
+        // Verify that the mockServerContext.setSequencerEpoch() was called and inspect the argument
+        ArgumentCaptor<Long> longArgumentCaptor = ArgumentCaptor.forClass(Long.class);
+        verify(mockServerContext).setSequencerEpoch(longArgumentCaptor.capture());
+
+        // Assert that serverContext.setSequencerEpoch() argument is equal to the bootstrapMsgEpoch value
+        assertThat(longArgumentCaptor.getValue()).isEqualTo(bootstrapMsgEpoch);
+        // Assert that sequencerEpoch value is updated to the bootstrapMsgEpoch value passed in the request
+        assertThat(sequencerServer.getSequencerEpoch()).isEqualTo(bootstrapMsgEpoch);
+        // Assert that globalLogTail value is updated to the globalTail value passed in the request
+        assertThat(sequencerServer.getGlobalLogTail()).isEqualTo(globalTail);
+        // Assert that epochRangeLowerBound value is updated from its initial value Layout.INVALID_EPOCH
+        assertThat(sequencerServer.getEpochRangeLowerBound()).isEqualTo(bootstrapMsgEpoch);
+
+        ArgumentCaptor<Map<UUID, StreamAddressSpace>> streamAddressSpaceMapCaptor = ArgumentCaptor.forClass(HashMap.class);
+        verify(mockStreamAddressSpaceMap).clear();
+        verify(mockStreamAddressSpaceMap).putAll(streamAddressSpaceMapCaptor.capture());
+        Map<UUID, StreamAddressSpace> capturedStreamAddressSpace = streamAddressSpaceMapCaptor.getValue();
+        assertThat(capturedStreamAddressSpace.size()).isEqualTo(1);
+        assertThat(capturedStreamAddressSpace).containsKey(streamA);
+
+        StreamAddressSpace responseStreamAddressSpace = capturedStreamAddressSpace.get(streamA);
+        assertThat(response.getPayload().getStreamsAddressResponse().getLogTail()).isEqualTo(globalTail);
+        assertThat(responseStreamAddressSpace.getAddressMap()).isEqualTo(Roaring64NavigableMap.bitmapOf(globalTail));
+        assertThat(responseStreamAddressSpace.getTrimMark()).isEqualTo(Address.NON_ADDRESS);
+    }
+
+    /**
+     * Test that the SequencerServer correctly handles a StreamsAddressRequestMsg.
+     * <p>
+     * If the request type is StreamsAddressRequestMsg.STREAMS then the server should respond with
+     * the address map of a particular stream requested by the client.
+     * <p>
+     * If the request type is StreamsAddressRequestMsg.ALL_STREAMS then the server should respond with
+     * the address map of all the streams.
+     * <p>
+     * If the request type is StreamsAddressRequestMsg.INVALID then the server should throw a new
+     * {@link IllegalArgumentException}.
+     */
+    @Test
+    public void testStreamAddressRequest() {
```

Comment:
Can we consider breaking up these larger tests into smaller units, instead of testing all the data paths for an RPC in a single unit test?

Reply:
They all operate on the same `streamsAddressSpaceMap` data on the sequencer server. The only difference between each one is the way they fetch the address space data (specific stream vs all stream). The third one is an invalid one. Hence I clubbed them. Splitting would slightly increase the time and space required for tests.